code/modules/ninja/suit/ninjaDrainAct.dm

```diff
@@ -1,263 +1,321 @@
+	if(!ninja_suit || !ninja || !ninja_gloves)
 		return INVALID_DRAIN
 
-	var/maxcapacity = 0 //Safety check
+	if(ninja_gloves.cyborg_hijacks > 0)
+		to_chat(src, "<span class='danger'>Warni-***BZZZZZZZZZRT*** UPLOADING SPYDERPATCHER VERSION 9.5.2...</span>")
+		if (do_after(ninja, 60, target = src))
+			ninja_gloves.cyborg_hijacks--
+			spark_system.start()
+			playsound(loc, "sparks", 50, TRUE)
+			to_chat(src, "<span class='danger'>UPLOAD COMPLETE.  NEW CYBORG MODULE DETECTED.  INSTALLING...</span>")
+			faction = list(ROLE_NINJA)
+			bubble_icon = "syndibot"
+			lawupdate = FALSE
+			scrambledcodes = TRUE
+			ionpulse = TRUE
+			laws = new /datum/ai_laws/ninja_override()
+			module.transform_to(pick(/obj/item/robot_module/syndicate, /obj/item/robot_module/syndicate_medical, /obj/item/robot_module/saboteur))
+			
+			var/datum/antagonist/ninja/ninja_antag = ninja.mind.has_antag_datum(/datum/antagonist/ninja)
+			if(!ninja_antag)
+				return
+			var/datum/objective/cyborg_hijack/objective = locate() in ninja_antag.objectives
+			if(objective)
+				objective.completed = TRUE
+		return
+	
+	var/maxcapacity = FALSE //Safety check
 	var/drain = 0 //Drain amount
-	. = 0
+	var/drain_total = 0
 
 	to_chat(src, "<span class='danger'>Warning: Unauthorized access through sub-route 12, block C, detected.</span>")
 
 	if(cell && cell.charge)
-		while(G.candrain && cell.charge > 0 && !maxcapacity)
-			drain = rand(G.mindrain,G.maxdrain)
+		while(ninja_gloves.candrain && cell.charge > 0 && !maxcapacity)
+			drain = rand(ninja_gloves.mindrain, ninja_gloves.maxdrain)
 			if(cell.charge < drain)
 				drain = cell.charge
-			if(S.cell.charge+drain > S.cell.maxcharge)
-				drain = S.cell.maxcharge - S.cell.charge
-				maxcapacity = 1
-			if (do_after(H,10))
+			if(ninja_suit.cell.charge + drain > ninja_suit.cell.maxcharge)
+				drain = ninja_suit.cell.maxcharge - ninja_suit.cell.charge
+				maxcapacity = TRUE
+			if(do_after(ninja ,10))
 				spark_system.start()
 				playsound(loc, "sparks", 50, TRUE)
 				cell.use(drain)
-				S.cell.give(drain)
-				. += drain
+				ninja_suit.cell.give(drain)
+				drain_total += drain
 			else
 				break
 
-
-//CARBON MOBS//
-/mob/living/carbon/ninjadrain_act(obj/item/clothing/suit/space/space_ninja/S, mob/living/carbon/human/H, obj/item/clothing/gloves/space_ninja/G)
-	if(!S || !H || !G)
-		return INVALID_DRAIN
-
-	. = DRAIN_MOB_SHOCK_FAILED
-
-	//Default cell = 10,000 charge, 10,000/1000 = 10 uses without charging/upgrading
-	if(S.cell && S.cell.charge && S.cell.use(1000))
-		. = DRAIN_MOB_SHOCK
-		//Got that electric touch
-		var/datum/effect_system/spark_spread/spark_system = new /datum/effect_system/spark_spread()
-		spark_system.set_up(5, 0, loc)
-		playsound(src, "sparks", 50, TRUE)
-		visible_message("<span class='danger'>[H] electrocutes [src] with [H.p_their()] touch!</span>", "<span class='userdanger'>[H] electrocutes you with [H.p_their()] touch!</span>")
-		electrocute_act(25, H)
```

Comment:
Since I don't see any documentation on it, are the stungloves aspect of the ninja suit being removed or did it get rebuilt elsewhere in the code?
Assuming complete removal I ask whether the `DRAIN_MOB_SHOCK` and 'DRAIN_MOB_SHOCK_FAILED' defines over in __ninjaDefines.dm are still used else and still worth keeping.

Code stuff aside, I figured ninja gloves would see a hit eventually given the current views on one-click-stuns, yet would it still be reasonable to rework it to share stunprod mechanics and take two shocks to incapacitate someone as space ninjas have little else in the way of non-lethal takedowns.

Similar thoughts for the energy net jail and whether it could return abductees much like those taken by syndicate contractors.

Reply:
I don't believe DRAIN_MOB_SHOCK or DRAIN_MOB_SHOCK_FAILED are used anywhere, I'll look to remove them later.  As of right now, you still do have two options of non-lethal takedown, the net and the new ninja stars, which drain stamina once attached to their target and they move around.  I will consider having the net work like the contractor's capture, but I'm still inclined to keeping it out as it is right now.